lib/Twig/Cache/Filesystem.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) 2015 Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Implements a cache on the filesystem.
+ *
+ * @author Andrew Tch <andrew@noop.lv>
+ */
+class Twig_Cache_Filesystem implements Twig_CacheInterface
+{
+    private $directory;
+
+    /**
+     * @param $directory string The root cache directory
+     */
+    public function __construct($directory)
+    {
+        $this->directory = $directory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCacheKey($className, $prefix)
+    {
+        $class = substr($className, strlen($prefix));
+
+        return $this->directory.'/'.$class[0].'/'.$class[1].'/'.$class.'.php';
```

Comment:
what if ``$class`` is shorter than 2 chars ? this code should be safe about what it receives